pkg/prober/intrumentation.go

```diff
@@ -23,7 +24,8 @@ type InstrumentationProbe struct {
 	component component.Component
 	logger    log.Logger
 
-	status *prometheus.GaugeVec
+	status       *prometheus.GaugeVec
+	formerStatus string
```

Comment:
We need a lock here since Ready/NotReady could be called from multiple goroutines.